test/performance/alphabet/alphabet_assign_char_benchmark.cpp

```diff
@@ -53,4 +60,30 @@ BENCHMARK_TEMPLATE(assign_char_, seqan3::masked<seqan3::dna4>);
 BENCHMARK_TEMPLATE(assign_char_, seqan3::qualified<seqan3::dna4, seqan3::phred42>);
 BENCHMARK_TEMPLATE(assign_char_, seqan3::qualified<seqan3::dna5, seqan3::phred63>);
 
+#if SEQAN3_HAS_SEQAN2
+template <typename alphabet_t>
+void assign_char_2(benchmark::State & state)
+{
+    std::array<char, 256> chars{};
+    size_t i = 0;
```

Comment:
The maximum value of `i` is 256, so a smaller type is sufficient. 
Why don't you use `std::iota` for filling the array with sequentially increasing values?